docs/website/docs/user-guides/advanced/connecting-a-service.md

```diff
@@ -0,0 +1,229 @@
+
+From a new terminal, run the following command that will add necessary data to devfile.yaml:
+```shell
+odo add binding \
+  --service mongodb-instance/PerconaServerMongoDB \
+  --name my-go-app-mongodb-instance \
+  --bind-as-files=false
+```
+```shell
+$ odo add binding --service mongodb-instance/PerconaServerMongoDB --name my-go-app-mongodb-instance --bind-as-files=false
+ ✓  Successfully added the binding to the devfile.
+Run `odo dev` to create it on the cluster.
+```
+
+:::note
+`--binding-as-files=false` because our code logic relies on obtaining environment variables from the system instead of reading data from files.
+:::
+
+Wait for `odo dev` to detect the new changes to the devfile.yaml. 
+```shell
+$ odo dev
+  __                                                                                                                                              
+ /  \__     Developing using the my-go-app Devfile                                                                                                
+ \__/  \    Namespace: my-percona-server-mongodb-operator                                                                                         
+ /  \__/    odo version: v3.0.0-rc1
+ \__/
+
+...
+...
+...
+Pushing files...
+
+
+File /tmp/go/devfile.yaml changed
+ •  Waiting for Kubernetes resources  ...
+ ✓  Creating kind ServiceBinding 
+Error occurred on Push - watch command was unable to push component: some servicebindings are not injected
+
+Updating Component...
+
+ •  Waiting for Kubernetes resources  ...
+Error occurred on Push - watch command was unable to push component: some servicebindings are not injected
+
+ ⚠  Pod is Terminating
+ •  Waiting for Kubernetes resources  ...
+ ✗  Finished executing the application (command: run) [1m]
+ ⚠  No pod exists
+ ⚠  Pod is Pending
+ ✓  Pod is Running
+ ✓  Syncing files into the container [170ms]
+ ✓  Building your application in container on cluster (command: build) [192ms]
+ •  Executing the application (command: run)  ...
+ -  Forwarding from 127.0.0.1:40001 -> 8080
+                                                                                                                                                  
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+Pushing files...
+
+
+File /tmp/go/devfile.yaml changed
+
+File /tmp/go/.odo/devstate.json changed
+ •  Waiting for Kubernetes resources  ...
+ ✓  Syncing files into the container [1ms]
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+
+```
+
+### Check the connection again
+Once it is done, call the URL again. We should now be connected.
+```shell
+curl 127.0.0.1:40001
+```
+```shell
+$ curl 127.0.0.1:40001
+Successfully connected and pinged.
+```
+
+### Exit and cleanup
+Press `Ctrl+c` to exit `odo dev`.
```

Comment:
This is already in the output? Perhaps we don't need this in the doc?

Reply:
I think it's alright to be explicit here before user proceeds with the next step.